src/components/post/postList/DeletePost.js

```diff
@@ -43,17 +40,11 @@ const DeletePost = ({
         >
           <DeleteOutline color='secondary' />
         </CustomButton>
-        <Dialog open={open} onClose={handleClose} fullWidth maxWidth='sm'>
-          <DialogTitle>Are you sure you want to delete this post?</DialogTitle>
-          <DialogActions>
-            <Button onClick={handleClose} color='primary'>
-              Cancel
-            </Button>
-            <Button onClick={deletePost} color='secondary'>
-              Delete
-            </Button>
-          </DialogActions>
-        </Dialog>
+        <DeletePostDialog
+          open={open}
+          handleClose={handleClose}
```

Comment:
Technically, you could make this component return the CustomButton when !open, and move all the state onboard. You would end up passing in nothing but deletePost, or even just passing in post and let it handle the delete itself too.

Reply:
My only issue is that would increase the number of indentations in the component, which was the reason for the refactor.

Reply:
No I mean you can move the delete button and the "open" state barracks into your DeletePostDialog component. Just adding a usestate hook and then
if(!open){
Return custom button here
}
Before you return the actual modal.

Reply:
Back, not barracks

Reply:
Fixed